src/components/molecules/Chatbox/ChatboxMessageList.tsx

```diff
@@ -0,0 +1,171 @@
+                          }`}
+                          key={`${message.from}-${message.ts_utc}`}
+                        >
+                          <div className="chat-message-bubble">
+                            {getLinkFromText(message.text)}
+                          </div>
+                          <div className="chat-message-author">
+                            <img
+                              onClick={() => {
+                                setSelectedUserProfile({
+                                  ...usersById[message.from],
+                                  id: message.from,
+                                });
+                              }}
+                              key={`${message.from}-messaging-the-band`}
+                              className="chat-message-avatar"
+                              src={
+                                (!sender.anonMode && sender.pictureUrl) ||
+                                DEFAULT_PROFILE_IMAGE
+                              }
+                              title={
+                                (!sender.anonMode && sender.partyName) ||
+                                DEFAULT_PARTY_NAME
+                              }
+                              alt={`${
+                                (!sender.anonMode && sender.partyName) ||
+                                DEFAULT_PARTY_NAME
+                              } profile`}
+                              width={profileImageSize}
+                              height={profileImageSize}
+                            />
+                            <div className="chat-message-pseudo">
+                              {(!sender.anonMode && sender.partyName) ||
+                                DEFAULT_PARTY_NAME}{" "}
+                              <span className="timestamp">
+                                {formatUtcSeconds(message.ts_utc)}
+                              </span>
+                            </div>
+                            {allowDelete && (
+                              <div
+                                className="chat-message-delete"
+                                onClick={() => {
+                                  setMessageToDelete(message);
+                                  setShowDeleteModal(true);
+                                }}
+                              ></div>
+                            )}
+                          </div>
+                        </div>
+                      </>
+                    )}
+                  </React.Fragment>
+                );
+              }
+            )}
+        </div>
+      )}
+      {!messages.length && (
+        <div className="chat-messages-empty">
+          {emptyListMessage ?? "No private messages yet."}
+        </div>
+      )}
+      <UserProfileModal
+        userProfile={selectedUserProfile}
+        show={selectedUserProfile !== undefined}
+        onHide={() => setSelectedUserProfile(undefined)}
```

Comment:
This is a simple state setter it doesn't need useCallback ,React guarantees that the function identity is stable and won’t change on re-renders.

Reply:
React guarantees that `setSelectedUserProfile`'s function reference won't change, but the function reference of the wrapping function (`() =>`) is being created on every render.

Something like this would do the trick I believe:

```typescript
const hideUser = useCallback(() => {
  setSelectedUserProfile(undefined)
}, []);
```

Then passed to this like:

```typescript
// ..snip.. 
onHide={hideUser}
// ..snip.. 
```

Reply:
fixed in 70e8253242c4c56f3678a5035e0e6566912da044